### PR TITLE
EZP-31017: Added Password Validator to change password form

### PR DESCRIPTION
**Pull request by ViniTou**
| Question      | Answer
| ------------- | ---
| Tickets       |  https://jira.ez.no/browse/EZP-31017
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | yes (but not sure if it is covered by BC policy - mandatory option for FormType)
| Tests pass?   | yes
| Doc needed?   | no
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

As a part of removing circular dependencies between `ezplatform-user` and `ezplatform-admin-ui` I moved validators in question into this repository.

I have also deprecated passing contentType as part of data class as it should from start be passed as form option and removed deprecated routing aliases.

## Additional PRs:
https://github.com/ezsystems/ezplatform-content-forms/pull/24
https://github.com/ezsystems/ezplatform-admin-ui/pull/1348

#### Checklist:
- [x] Implement tests
- [x] Coding standards (`$ composer fix-cs`)

### CODE REVIEW

File: tests/lib/Validator/Constraint/PasswordTest.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+{
+    /** @var \EzSystems\EzPlatformUser\Validator\Constraints\Password */
+    private $constraint;
+
+    protected function setUp(): void
+    {
+        $this->constraint = new Password();
+    }
+
+    public function testConstruct()
```
Comment (alongosz) [inline tests/lib/Validator/Constraint/PasswordTest.php:26]: Could you use `: void` for test methods as well?
```suggestion
    public function testConstruct(): void
```

File: tests/lib/Validator/Constraint/PasswordTest.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    {
+        $this->assertSame('ez.user.password.invalid', $this->constraint->message);
+    }
+
+    public function testValidatedBy()
```
Comment (alongosz) [inline tests/lib/Validator/Constraint/PasswordTest.php:31]:
```suggestion
    public function testValidatedBy(): void
```

File: tests/lib/Validator/Constraint/PasswordTest.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    {
+        $this->assertSame(PasswordValidator::class, $this->constraint->validatedBy());
+    }
+
+    public function testGetTargets()
```
Comment (alongosz) [inline tests/lib/Validator/Constraint/PasswordTest.php:36]:
```suggestion
    public function testGetTargets(): void
```

File: tests/lib/Validator/Constraint/PasswordValidatorTest.php [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        $this->userService = $this->createMock(UserService::class);
+        $this->executionContext = $this->createMock(ExecutionContextInterface::class);
+        $this->validator = new PasswordValidator($this->userService);
+        $this->validator->initialize($this->executionContext);
+    }
+
+    /**
+     * @dataProvider dataProviderForValidateNotSupportedValueType
+     */
+    public function testValidateShouldBeSkipped($value)
```
Comment (alongosz) [inline tests/lib/Validator/Constraint/PasswordValidatorTest.php:43]:
```suggestion
    public function testValidateShouldBeSkipped($value): void
```

File: tests/lib/Validator/Constraint/PasswordValidatorTest.php [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+            ->method('validatePassword');
+
+        $this->executionContext
+            ->expects($this->never())
+            ->method('buildViolation');
+
+        $this->validator->validate($value, new Password());
+    }
+
+    public function testValid()
```
Comment (alongosz) [inline tests/lib/Validator/Constraint/PasswordValidatorTest.php:56]:
```suggestion
    public function testValid(): void
```

File: tests/lib/Validator/Constraint/PasswordValidatorTest.php [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+        $this->executionContext
+            ->expects($this->never())
+            ->method('buildViolation');
+
+        $this->validator->validate($password, new Password([
+            'contentType' => $contentType,
+        ]));
+    }
+
+    public function testInvalid()
```
Comment (alongosz) [inline tests/lib/Validator/Constraint/PasswordValidatorTest.php:81]:
```suggestion
    public function testInvalid(): void
```

File: tests/lib/Validator/Constraint/PasswordValidatorTest.php [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+
+    public function testValid()
+    {
+        $password = 'pass';
+        $contentType = $this->createMock(ContentType::class);
+
+        $this->userService
+            ->expects($this->once())
+            ->method('validatePassword')
+            ->willReturnCallback(function ($actualPassword, $actualContext) use ($password, $contentType) {
```
Comment (alongosz) [inline tests/lib/Validator/Constraint/PasswordValidatorTest.php:64]: Do `$actualPassword` and `$actualContext` have a strict type?

File: tests/lib/Validator/Constraint/PasswordValidatorTest.php [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+    {
+        $contentType = $this->createMock(ContentType::class);
+        $password = 'pass';
+        $errorParameter = 'foo';
+        $errorMessage = 'error';
+
+        $this->userService
+            ->expects($this->once())
+            ->method('validatePassword')
+            ->willReturnCallback(function ($actualPassword, $actualContext) use ($password, $contentType, $errorMessage, $errorParameter) {
```
Comment (alongosz) [inline tests/lib/Validator/Constraint/PasswordValidatorTest.php:91]: Same question as above

File: tests/lib/Validator/Constraint/PasswordTest.php [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformUser\Tests\Validator\Constraint;
+
+
```
Comment (adamwojs) [inline tests/lib/Validator/Constraint/PasswordTest.php:11]: Nitpick: 

```suggestion
```

**Pull request merged**